src/domain/chains/chains.repository.ts

```diff
@@ -47,6 +48,28 @@ export class ChainsRepository implements IChainsRepository {
     return valid;
   }
 
+  async getAllChains(): Promise<Array<Chain>> {
+    const chains: Array<Chain> = [];
+
+    let offset: number | undefined;
+
+    while (true) {
```

Comment:
I think we should be a bit more defensive here: `while(true)` seems scary, and we are relying on the fact the datasource (Config Service in this case) works as expected and does not have a bug affecting the `offset` value. 

What do you think about having some kind of threshold to limit the max number of iterations in the loop? We could define a max number of Chain objects that can be handled in total and divide it by the default `limit` path param.

Reply:
I've set an "upper request limit" (adjustable by env. var.) so that only n amount of subsequent requests are made. We also now fetch the upper limit to hopefully also prevent subsequent requests. As of 6b4f5f7a9a6b1484a514880a028c11374fdfb3a3.